### PR TITLE
Completed Week 6 Assessment

### PR DESCRIPTION
**Pull request by NikkiDunlap**
# Submitting Week 6 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [ X ] Coding Challenges
- [ X ] Coding Concepts
- [ X ] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -13,34 +13,170 @@
...
 // a) Create a test with an expect statement using the variable provided.
+  // Pseudocode:
+    // Given an array of objects as an input
+    // Create a describe function named "outcomeSentences" that takes an array as an input
+    // Use the .map() method to iterate over every object in the "people" array and to create a new array of sentences
+    // For each person, the .map() method is using split to put the string into an array of words
+    // We will use .toUpperCase to capitalize each first letter of the word
+    // We will use .join() method to join the words back into a string
+    // Map returns a string that combines the capitalized name with the occupation of that person in the sentence
+    // Function returns an array of sentences for each person in the original array with their name capitalized and occupation included
```
Comment (chrisyhlee) [inline code-challenges.test.js:24]: Some missing elements, such as no mention of .charAt or .slice, which is used in the code! Your pseudo code could definitely be shortened, even with the addition of the missing elements. Your function itself and logic is great!

File: code-challenges.test.js [hunk 2]
```diff
@@ -13,34 +13,170 @@
...
+          // Time:        0.171 s
+          // Ran all test suites matching /jest2.test.js/i.
+          // ✨  Done in 4.40s.
 
 // --------------------2) Create a function that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
+      // Pseudocode:
+        // Given an array with mixed datatypes
+        // Create describe function called "remaindersByThree"
+        // Create an empty array called "ourResult" to keep the remainder values for each number in the given array when it is divided by 3
+        // Use .toEqual() method to compare the actual output to the expected output that was given to us
```
Comment (chrisyhlee) [inline code-challenges.test.js:71]: No need to add the test to the pseudo code!

File: code-challenges.test.js [hunk 3]
```diff
@@ -13,34 +13,170 @@
...
+        // Use a "for" loop to iterate over each elemet in the given arrays
+        // Use typeof to check to see if it is a number or not
+        // If it is a number, the function will calculate the remainder using the (% 3) then it will add that value to the ourResult array using .push()
+        // The .push() method will add the new numbers to the end of the array, then we will be able to see the new length
+        // Function returns an array of only the remainders of the numbers when divided by 3
```
Comment (chrisyhlee) [inline code-challenges.test.js:76]: Great explanation of your function

File: code-challenges.test.js [hunk 4]
```diff
@@ -13,34 +13,170 @@
...
 
 // --------------------3) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
+  // Pseudocode:
+    // Given 2 variables 
+    // Create a describe function named "cubeAndSum"
+    // Use cubeAndSum1 & cubeAndSum2 arrays as arguments 
+    // Create 2 test cases using the given arrays
+    // Create an "it" function for each test case
+    // The first test case is testing to make sure that the output is equal to 99
+    // The second test case is testing to make sure the output is 1125
```
Comment (chrisyhlee) [inline code-challenges.test.js:128]: Same thing as the last two, any pseudo code above this can be left out!

File: rails-commenting.rb [hunk 5]
```diff
@@ -6,39 +6,39 @@
...
   end
 
-  # ---3)
+  # ---3) The show action is defined. In that, the BlogPost.find will allow us to retrieve a single blog post from BlogPost.all. The params[:id] let's us identify which blog post we want to retrieve.
   def show
     @post = BlogPost.find(params[:id])
   end
 
-  # ---4)
+  # ---4) The new action is defined. This gives us a new instance of the model BlogPost and puts it in @post. This new instance variable will hold a new blog post.
```
Comment (chrisyhlee) [inline rails-commenting.rb:21]: Remember the new method is a "GET" request to display a form for the user!

**Review by chrisyhlee — APPROVED** [review]
Last assessment complete, congratulations! This displays Advanced Beginner level. The main focus for you would be getting your pseudo code clear and concise, and removing the unnecessary parts. Overall, you have a solid understanding of the concepts you are using. The explanation of it though could be improved. You are doing awesome, keep up the hard work! Excited to see what you do on these upcoming projects!

File: code-challenges.test.js (hunk 1 above)
Comment (NikkiDunlap) [reply code-challenges.test.js:24]: Thanks a lot Chris! I always leave something out smh!

File: code-challenges.test.js (hunk 3 above)
Comment (NikkiDunlap) [reply code-challenges.test.js:76]: Thanks Chris!

**Pull request merged**